server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -3761,6 +3771,65 @@ public void testRestoreIncreasesPrimaryTerms() {
         }
     }
 
+    public void testSnapshotDifferentIndicesBySameName() {
```

Comment:
If we reduce the number of primaries, can this lead to orphaned shard folders when deleting the last snapshot that was referencing the prior index with the higher count of primaries?

Reply:
You're right ... this needs some special case handling now. Otherwise we write an empty index-${uuid} blob that isn't referencing anything now. Will fix that tomorrow.

This whole index-name business is getting pretty weird ... but I don't see a clean way of changing the lookup to be by index-uuid either so :(